plugins/CMakeLists.txt

```diff
@@ -2,186 +2,91 @@ cmake_minimum_required(VERSION 3.22...3.25)
-  else()
-    obs_status(FATAL_ERROR "obs-browser submodule not available.")
-  endif()
-endfunction()
-
-# APPLE/WIN32/UNIX are soft-deprecated: https://discourse.cmake.org/t/platform-id-vs-win32-vs-cmake-system-name/1226/2
-if(OS_WINDOWS)
-  add_subdirectory(coreaudio-encoder)
-  add_subdirectory(win-wasapi)
-  add_subdirectory(win-dshow)
-  add_subdirectory(win-capture)
-  add_subdirectory(decklink)
-  add_subdirectory(obs-qsv11)
-  add_subdirectory(obs-text)
-  add_subdirectory(vlc-video)
-  add_subdirectory(obs-vst)
-  add_subdirectory(nv-filters)
-
-  check_obs_browser()
-elseif(OS_MACOS)
-  add_subdirectory(coreaudio-encoder)
-  add_subdirectory(mac-avcapture)
-  add_subdirectory(mac-capture)
-  add_subdirectory(mac-videotoolbox)
-  add_subdirectory(mac-syphon)
-  add_subdirectory(mac-virtualcam)
-  add_subdirectory(decklink)
-  add_subdirectory(vlc-video)
-  add_subdirectory(linux-jack)
-  add_subdirectory(obs-vst)
-
-  check_obs_browser()
-elseif(OS_LINUX)
-  add_subdirectory(linux-capture)
-  add_subdirectory(linux-pulseaudio)
-  add_subdirectory(linux-v4l2)
-  add_subdirectory(linux-jack)
-  add_subdirectory(linux-alsa)
-  add_subdirectory(linux-pipewire)
-  add_subdirectory(decklink)
-  add_subdirectory(vlc-video)
-  add_subdirectory(sndio)
-  add_subdirectory(obs-vst)
-  add_subdirectory(obs-qsv11)
-
-  check_obs_browser()
-elseif(OS_FREEBSD)
-  add_subdirectory(linux-capture)
-  add_subdirectory(linux-pipewire)
-  add_subdirectory(linux-pulseaudio)
-  add_subdirectory(linux-v4l2)
-  add_subdirectory(linux-jack)
-  add_subdirectory(linux-alsa)
-  add_subdirectory(vlc-video)
-  add_subdirectory(oss-audio)
-  add_subdirectory(sndio)
-
-  obs_status(STATUS "obs-browser plugin not available.")
-  obs_status(STATUS "obs-vst plugin not available.")
-elseif(OS_OPENBSD)
-  add_subdirectory(linux-capture)
-  add_subdirectory(sndio)
-
-  obs_status(STATUS "obs-browser plugin not available.")
-  obs_status(STATUS "obs-vst plugin not available.")
-endif()
-
-if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/obs-websocket/CMakeLists.txt)
-  add_subdirectory(obs-websocket)
-else()
-  obs_status(FATAL_ERROR "obs-websocket submodule not available.")
-endif()
-
-add_subdirectory(image-source)
-add_subdirectory(obs-x264)
-add_subdirectory(obs-libfdk)
-add_subdirectory(obs-ffmpeg)
-add_subdirectory(obs-outputs)
-add_subdirectory(obs-filters)
-add_subdirectory(obs-transitions)
-add_subdirectory(rtmp-services)
-add_subdirectory(text-freetype2)
-add_subdirectory(aja)
-add_subdirectory(obs-webrtc)
+)
+add_obs_plugin(obs-webrtc)
+
+check_obs_websocket()
+
+add_obs_plugin(obs-x264)
+add_obs_plugin(oss-audio PLATFORMS FREEBSD OPENBSD)
+add_obs_plugin(rtmp-services)
+add_obs_plugin(sndio PLATFORMS LINUX FREEBSD OPENBSD)
+add_obs_plugin(text-freetype2)
+add_obs_plugin(vlc-video WITH_MESSAGE)
+add_obs_plugin(win-capture PLATFORMS WINDOWS)
+add_obs_plugin(win-dshow PLATFORMS WINDOWS)
+add_obs_plugin(win-wasapi PLATFORMS WINDOWS)
+
+return()
```

Comment:
As with the other review note, I'm not sure that `return()` is necessary now that this is not inside an `if()`, but @PatTheMav would likely know for sure.